gpt_index/utils.py

```diff
@@ -50,3 +50,42 @@ def get_new_id(d: Set) -> str:
         if new_id not in d:
             break
     return new_id
+
+
+def llm_token_counter(method_name_str: str) -> Callable:
+    """
+    Use this as a decorator for methods in index/query classes that make calls to LLMs.
+
+    At the moment, this decorator can only be used on class instance methods with a
+    `_llm_predictor` attribute.
+
+    Do not use this on abstract methods.
+
+    For example, consider the class below:
+    ```
+    class GPTTreeIndexBuilder:
+        ...
+        @llm_token_counter("build_from_text")
+        def build_from_text(self, documents: Sequence[BaseDocument]) -> IndexGraph:
+            ...
+    ```
+
+    If you run `build_from_text()`, it will print the output in the form below:
+
+    ```
+    [build_from_text] Total token usage: <some-number> tokens
+    ```
+    """
+
+    def wrap(f: Callable) -> Callable:
+        def wrapped_f(_self: Any, *args: Any, **kwargs: Any) -> Any:
```

Comment:
nit: would you be able to name this something like `wrapped_llm_predict` for clarity

Reply:
yup! done: https://github.com/jerryjliu/gpt_index/pull/85/commits/21d3a21748534c5b51fa216df30509843250ed2a